src/SOS/SOS.UnitTests/SOSRunner.cs

```diff
@@ -844,37 +884,64 @@ public async Task LoadSosExtension()
                 if (_config.IsDesktop)
                 {
                     // Force the desktop sos to be loaded and then unload it.
-                    commands.Add(".cordll -l");
-                    commands.Add(".unload sos");
+                    if (!string.IsNullOrEmpty(runtimeSymbolsPath))
+                    {
+                        commands.Add($".cordll -lp {runtimeSymbolsPath}");
+                    }
+                    else
+                    {
+                        commands.Add(".cordll -l");
+                    }
                 }
+                commands.Add(".unload sos");
                 commands.Add($".load {sosPath}");
                 commands.Add(".reload");
                 commands.Add(".chain");
                 if (!string.IsNullOrEmpty(setHostRuntime))
                 {
                     commands.Add($"!SetHostRuntime {setHostRuntime}");
                 }
-                // Add the path to runtime so SOS can find DAC/DBI for triage dumps
-                if (_dumpType.HasValue && _dumpType == DumpType.Triage)
+                // If there is no host runtime and a single-file app or a triage dump, add the path to runtime so SOS can find DAC/DBI.
+                if ((isHostRuntimeNone && _config.PublishSingleFile) || 
+                    (_dumpType.HasValue && _dumpType.Value == DumpType.Triage))
                 {
-                    string runtimeSymbolsPath = _config.RuntimeSymbolsPath;
-                    if (runtimeSymbolsPath != null)
+                    if (!string.IsNullOrEmpty(runtimeSymbolsPath))
                     {
-                        commands.Add("!SetClrPath " + runtimeSymbolsPath);
+                        commands.Add($"!SetClrPath {runtimeSymbolsPath}");
                     }
                 }
+                if (!isHostRuntimeNone && !string.IsNullOrEmpty(setSymbolServer))
+                {
+                    commands.Add($"!SetSymbolServer {setSymbolServer}");
+                }
                 break;
             case NativeDebugger.Lldb:
                 commands.Add($"plugin load {sosPath}");
                 if (!string.IsNullOrEmpty(setHostRuntime))
                 {
                     commands.Add($"sethostruntime {setHostRuntime}");
                 }
+                // If there is no host runtime and a single-file app, add the path to runtime so SOS can find DAC/DBI.
+                if (isHostRuntimeNone && _config.PublishSingleFile)
```

Comment:
Should we let this fall back to native runtime enumeration? This would help us test if WinDBG can load the symbols